data_tools/buildhub/utils.py

```diff
@@ -0,0 +1,93 @@
+from os.path import basename
+
+
+def build_record_from_url(url):
+    record = {}
+    parts = basename(url).split('.')
+    url_parts = url.split('/')
+
+    product_name, major_version = parts[0].split('-')
+    channel = 'release'
+
+    if 'nightly' in url:
+        version = '{}.{}'.format(major_version, parts[1])
+        locale = parts[2]
+        platform = parts[3]
+        if 'aurora' in url:
+            channel = 'aurora'
+        else:
+            channel = 'nightly'
+    elif 'candidates' in url:
+        candidate_number = url_parts[7].strip('build')
+        version = '{}.{}rc{}'.format(major_version, parts[1], candidate_number)
+        locale = url_parts[9]
+        platform = url_parts[8]
+    else:
+        version = url_parts[6]
+        locale = url_parts[8]
+        platform = url_parts[7]
+        if 'esr' in version:
+            channel = 'esr'
+        elif 'b' in version:
+            channel = 'beta'
+
+    record['target'] = {
+        'product': product_name,
+        'version': version,
+        'platform': platform,
+        'locale': locale,
+        'channel': channel,
+    }
+    record["download"] = {"url": url}
+    record_id = build_record_id(record)
+    record['id'] = record_id
+    return record
+
+
+def build_record_id(record):
+    version = record["target"]["version"]
+    if record["target"]["channel"] == 'nightly':
+        url_parts = record["download"]["url"].split('/')
+        date_parts = url_parts[8].split('-')
+        date = '-'.join(date_parts[:6])
+        version = '{}_{}'.format(date, version)
+    return '{target[product]}_{version}_{target[platform]}_{target[locale]}'.format(
+        target=record['target'], version=version).replace('.', '-').lower()
+
+
+def build_record_id_from_url(url):
```

Comment:
The goal I had in mind with this PR was simply to generate the ID from the URL.
You asked me in a previous review to use the record rather the URL, but the current record doesn't contains the mandatory information